### DISCUSSION TITLE
Dealing with one-off transactions when importing from CSV

### DISCUSSION THREAD

**Discussion by gleber** [apauley/hledger-flow#116] [Q&A] [answered]
I am importing CSV files from a lot of places. Especially for my credit card, I often have one-off transactions which are somehow unique, and I want to handle them differently.

For example in case if I have paid for a one-off notary service for a real estate transaction using my credit card, I'd like these expenses to be added to a separate expenses subaccount which is specific to that real estate.

Is my best option is to have a one-off `if` block in the rules file, which will match just this one CSV line? This feels wrong, since the rules file becomes polluted with many one-off ifs like that which correspond to all such cases which happened throughout this year.

Thank you for `hledger-flow`! It is an amazing tool :)

**Comment by apauley** [comment, marked as answer]
Many of my rules are general, but like you I also have many one-off cases where I need a rule that matches just a single transaction.

For some of my accounts I've grouped them in a separate file, which I then `include` in the rules file of that account.

In this example it is `mybank-special.rules`:

```hledger
# Saved as: import/john/mybank/checking/mybank-checking.rules
include ../../../mybank-shared.rules
include mybank-special.rules
account1 assets:Current:John:MyBank:Checking
```

I guess you can break it down further by year if you need to.

**Reply by gleber** [reply]
Thank you! Sounds like this is one of those aspects of keeping your finances categorized where an UI application would excel. Having said that, I'd choose the openness of the plain-text approach any day of the week :)
